health-checker/src/main/java/io/stargate/health/DataStoreHealthChecker.java

```diff
@@ -22,61 +22,46 @@
 import io.stargate.db.datastore.Row;
 import java.util.UUID;
 import java.util.concurrent.Future;
-import org.osgi.framework.BundleContext;
-import org.osgi.framework.ServiceReference;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class DataStoreHealthChecker extends HealthCheck {
   private static final Logger logger = LoggerFactory.getLogger(DataStoreHealthChecker.class);
 
-  private final BundleContext context;
+  private final DataStoreFactory dataStoreFactory;
 
-  public DataStoreHealthChecker(BundleContext context) {
-    this.context = context;
+  public DataStoreHealthChecker(DataStoreFactory dataStoreFactory) {
+    this.dataStoreFactory = dataStoreFactory;
   }
 
   @Override
   protected Result check() {
     try {
-      ServiceReference<DataStoreFactory> dataStoreFactoryReference =
```

Comment:
Why are we removing this? The idea is to check that we can properly pull the bundle from the context and then exercise it.

Reply:
I've refactored it to make an activator "checking" that:
```
  private ServicePointer<DataStoreFactory> dataStoreFactory =
      ServicePointer.create(DataStoreFactory.class);
```
If the data store won't be available, it will detect that sooner. Extraction of `DataStore` is validated in the `*Activator` instead of health-checks. Isn't it enough?